cub/util_device.cuh

```diff
@@ -364,7 +364,7 @@ CUB_RUNTIME_FUNCTION inline cudaError_t PtxVersionUncached(int& ptx_version)
 
     // This is necessary for unused variable warnings in host compilers. The
     // usual syntax of (void)empty_kernel; was not sufficient on MSVC2015.
-    (void)reinterpret_cast<void*>(empty_kernel);
+    (void)reinterpret_cast<void*>(&empty_kernel);
```

Comment:
As I understand the original issue is related to:

```cuda
cudaFuncGetAttributes(&empty_kernel_attrs, empty_kernel))
// -->
cudaFuncGetAttributes(&empty_kernel_attrs, reinterpret_cast<void*>(empty_kernel)))
```

Here's [the reproducer](https://cuda.godbolt.org/z/zshYzfYv4). Somehow `cudaFuncGetAttributes` doesn't emit the warning. 

Has `(void)reinterpret_cast<void*>(&empty_kernel);` fixed any warnings in your case? If not, I suggest we leave this line as it is and cast the pointer explicitly in `cudaFuncGetAttributes`.

Reply:
Ah, the line number have changed since that bug was reported, and the link is currently pointing to the `reinterpret_cast`. Thanks for spotting this!

I'll update this.